### PR TITLE
[SPARK-30956][SQL][TESTS] Use intercept instead of try-catch to assert failures in IntervalUtilsSuite

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

In this PR, I addressed the comment from https://github.com/apache/spark/pull/27672#discussion_r383719562 to use `intercept` instead of `try-catch` block to assert  failures in the IntervalUtilsSuite

### Why are the changes needed?

improve tests
### Does this PR introduce any user-facing change?

no

### How was this patch tested?

Nah

### CODE REVIEW

**Comment by maropu** [comment]
I feel its better to file jira as improvements.

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/IntervalUtilsSuite.scala [hunk 1]
```diff
@@ -215,13 +172,9 @@ class IntervalUtilsSuite extends SparkFunSuite with SQLHelper {
...
 
-    try {
-      duration(Integer.MAX_VALUE + " month", TimeUnit.SECONDS, 31)
-      fail("Expected to throw an exception for the invalid input")
-    } catch {
-      case e: ArithmeticException =>
-        assert(e.getMessage.contains("overflow"))
-    }
+    val e =
+      intercept[ArithmeticException](duration(Integer.MAX_VALUE + " month", TimeUnit.SECONDS, 31))
```
Comment (maropu) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/IntervalUtilsSuite.scala]: nit: 
```
    val e = intercept[ArithmeticException] { 
      duration(Integer.MAX_VALUE + " month", TimeUnit.SECONDS, 31)
    }
```
?

**Comment by yaooqinn** [comment]
@maropu thanks for your suggestion, I have filed and linked the Jira.

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/IntervalUtilsSuite.scala (hunk 1 above)
Comment (yaooqinn) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/IntervalUtilsSuite.scala]: done, thanks.

**Comment by yaooqinn** [comment]
retest this please

**Review by kiszk — APPROVED** [review]

**Review by maropu — APPROVED** [review]
cc: @cloud-fan

**Review by cloud-fan — APPROVED** [review]
thanks for cleaning it up!

**Comment by maropu** [comment]
Thanks! Merged to master/3.0.

**Pull request closed without merging**